main/util/src/mill/util/Jvm.scala

```diff
@@ -14,10 +14,173 @@ import os.CommandResult
 
 object Jvm extends CoursierSupport {
 
+  /**
+   * Runs a JVM subprocess with the given configuration and returns a
+   * [[os.CommandResult]] with it's aggregated output and error streams.
+   *
+   * Exit code is not checked.
+   * Use [[ProcessUtil.toResult]] to do a proper check
+   *
+   * @param mainClass The main class to run
+   * @param classPath The classpath
+   * @param javaHome Optional Java Home override
+   * @param jvmArgs Arguments given to the forked JVM
+   * @param useCpPassingJar When `false`, the `-cp` parameter is used to pass the classpath
+   *                        to the forked JVM.
+   *                        When `true`, a temporary empty JAR is created
+   *                        which contains a `Class-Path` manifest entry containing the actual classpath.
+   *                        This might help with long classpaths on OS'es (like Windows)
+   *                        which only supports limited command-line length
+   * @param mainArgs Args passed to the `mainClass` main method
+   * @param env Environment variables used when starting the forked JVM
+   * @param propagateEnv If `true` then the current process' environment variables are propagated to subprocess
+   * @param cwd The working directory to be used by the forked JVM
+   * @param stdin Standard input
+   * @param stdout Standard output
+   * @param stderr Standard error
+   * @param mergeErrIntoOut If `true` then the error output is merged into standard output
+   */
+  def call(
+      mainClass: String,
+      classPath: Iterable[os.Path],
+      javaHome: Option[os.Path] = None,
+      jvmArgs: Seq[String] = Seq.empty,
+      useCpPassingJar: Boolean = false,
+      mainArgs: Seq[String] = Seq.empty,
+      env: Map[String, String] = Map.empty,
+      propagateEnv: Boolean = true,
+      cwd: os.Path = null,
+      stdin: os.ProcessInput = os.Inherit,
+      stdout: ProcessOutput = os.Inherit,
+      stderr: ProcessOutput = os.Inherit,
+      mergeErrIntoOut: Boolean = false,
+      destroyOnExit: Boolean = true
+  )(implicit ctx: Ctx): CommandResult = {
```

Comment:
Let's get rid of the `ctx: Ctx` parameter here and below in `spawn`, and also get rid of the `ctx.log.debug` calls. The occasional benefit of turning on debug logging probably isn't worth the complexity this adds to the signature and calling context